stream-chat-android-client/src/main/java/io/getstream/chat/android/client/uploader/ProgressTracker.kt

```diff
@@ -0,0 +1,45 @@
+package io.getstream.chat.android.client.uploader
+
+import io.getstream.chat.android.client.errors.ChatError
+import io.getstream.chat.android.client.utils.ProgressCallback
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+public class ProgressTracker {
+
+    private var numberOfLaps = 1
+    private var currentProgress: MutableStateFlow<Int> = MutableStateFlow(0)
```

Comment:
`currentLapProgress`? (or `currentItemProgress`, if the suggestion above is applied)